lib/db.js

```diff
@@ -512,19 +514,30 @@ module.exports = (
       uaDeviceType: token.uaDeviceType,
       lastAccessTime: token.lastAccessTime,
       createdAt: token.createdAt
-    }]
+    }
     let sessionTokens = []
+    return getGeoData(ip)
+    .then((res) => {
+      const state = res.location && res.location.state
+      const country = res.location && res.location.country
+      newToken.location = {state, country}
+    })
+    .catch(err => {
+      newToken.location = null
+    })
     // get the array of session tokens associated with the given uid
-    return this.redis.getAsync(uid)
+    .then(() => this.redis.getAsync(uid))
     .then(res => {
       if (res) {
         res = JSON.parse(res)
         // remove the token that we want to update from the array
-        const filteredRes = res.filter(tok => tok.tokenId !== token.id)
-        sessionTokens = sessionTokens.concat(filteredRes)
+        const filteredTokens = res.filter(tok => tok.tokenId !== token.id)
+        sessionTokens = sessionTokens.concat(filteredTokens)
       }
-      // add new updated token into array, and set the resulting array as the new value
-      sessionTokens = sessionTokens.concat(newToken)
+    })
+    // add new updated token into array, and set the resulting array as the new value
+    .then(() => {
+      sessionTokens.push(newToken)
       return this.redis.setAsync(uid, JSON.stringify(sessionTokens))
```

Comment:
same here, also needs to guard https://github.com/mozilla/fxa-auth-server/pull/1993/files#r129374585 , unless we have something that does this already